filter_plugins/oo_filters.py

```diff
@@ -229,7 +227,13 @@ def oo_filter_list(data, filter_attr=None):
             raise errors.AnsibleFilterError("|failed expects filter_attr is a str")
 
         # Gather up the values for the list of keys passed in
-        return [x for x in data if x[filter_attr]]
+        return [x for x in data if (filter_attr in x.keys()) and (x[filter_attr])]
```

Comment:
I added this because of a strange behaviour in ansible when it fails to evaluate an expression, it considers it as literraly.
This was the case for the options.infra_node expression that was not fully implemented when I started my dev on this PR.

Hence, unusable added me an entry in my oo_nodes_to_config which was not a real node.

This check allows to workaround such behaviours.
I don't have any strong argument to remove it nor to keep it.

Less head breaking to keep it, more frustrating to remove it.